pkg/controllers/provisioning/scheduling/topology_test.go

```diff
@@ -1311,20 +1310,26 @@ var _ = Describe("Topology", func() {
 				NodeTaintsPolicy:  &honor,
 			}}
 
-			ExpectApplied(ctx, env.Client, nodePool)
-			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov,
-				test.UnschedulablePods(test.PodOptions{
-					ObjectMeta: metav1.ObjectMeta{Labels: labels},
-					ResourceRequirements: v1.ResourceRequirements{
-						Requests: v1.ResourceList{
-							v1.ResourceCPU: resource.MustParse("1"),
-						},
+			pods := lo.Flatten(lo.Map(nodePools, func(np *v1beta1.NodePool, _ int) []*v1.Pod {
+				return test.UnschedulablePods(test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: labels,
 					},
 					TopologySpreadConstraints: topology,
-				}, 5)...,
-			)
-			// and should schedule all of the pods on the same node
-			ExpectSkew(ctx, env.Client, "default", &topology[0]).To(ConsistOf(5))
+					Tolerations: []v1.Toleration{{
```

Comment:
This may work, but I think the test will be clearer with:
- 1 node pool with a value for a particular label and taints
- 1 node pool with a different value for that label and not taints

and then schedule multiple pods with no tolerations and a required topology spread over the label, one should schedule against the non-tainted node pool and the others remain pending.

A second test has identical setup but taints policy is ignore and all pods schedule against the non-tainted node pool since they don't discover the tainted node pool.

Reply:
That makes sense and IIUC I think that's more or less what you had originally for this test (+ the first set of pods). I modified the original test to catch a case I had missed when I originally implemented the filter for which domains a pod should consider based on taints. Basically I mistakenly thought I could just do a set intersection of the taints for all of the nodepools, which would result in a pod with no tolerations considering `foo` as a domain in this case (instead of needing to tolerate `nodepool-1` or `nodepool-2`. To catch this I needed to have multiple NodePools with distinct taints that shared at least 1 domain. I'd be happy to add in an additional test, I agree that this one isn't super clear.

Reply:
Sure that works, let's add the other test and keep this as well.